### ISSUE TITLE
Failure to save multiple OneToOne relationships

### ISSUE THREAD

**Issue by vaishraj** [Impetus/kundera#312]
Due to a bug in FlushManager, only a single One-to-one relationship is picked by the code. 

Instead of 

``` java
            // Process One-To-One children
            for (NodeLink nodeLink : oneToOneChildren.keySet())
            {
                if (!node.isTraversed())
                {
                    // Push this node to stack
                    node.setTraversed(true);
                    stackQueue.push(node);
                    logEvent(node, eventType);

                    // Process child node Graph recursively
                    Node childNode = children.get(nodeLink);

                    addNodesToFlushStack(childNode, eventType);
                }
         }
```

The code should be

``` java
            // Process One-To-One children
            for (NodeLink nodeLink : oneToOneChildren.keySet())
            {
                if (!node.isTraversed())
                {
                    // Push this node to stack
                    node.setTraversed(true);
                    stackQueue.push(node);
                    logEvent(node, eventType);
               }
                    // Process child node Graph recursively
                    Node childNode = children.get(nodeLink);

                    addNodesToFlushStack(childNode, eventType);
           }
```

**Comment by mevivs** [comment]
:+1:  for change.

Can you please contribute this change along with a junit?

-Vivek

**Comment by vaishraj** [comment]
Hi Vivek :

Thanks. Looks like the base unit test itself is missing for the function,
so it will take me a while to add the unit test for the change. It looks
more complex than the change itself and I need to learn a bit more about
the Node and NodeType classes :).

I will try to get the unit test added to the FlushStackTest by tomorrow. I
have tested the change on my local class that needed and it's working.

-Raj

**Comment by mevivs** [comment]
Hi Raj,
That's ok. I will add a test case and verify this change. Thanks for your contribution.

-Vivek

**Comment by impetus-opensource-admin** [comment]
Fix has been added for this. Please verify.

**Comment by mevivs** [comment]
Fixed and releasing with 2.6. Please verify.
-Vivek
